src/main/java/com/invoiced/entity/AbstractEntity.java

```diff
@@ -324,7 +405,9 @@ public void delete() throws EntityException {
 		return entities;
 	}
 
-	abstract long getEntityId();
+	abstract long getEntityId() throws EntityException;
```

Comment:
To save code I think you can implement both of these getEntity methods on the abstract class to always throw an exception. Then in the child class it would be overridden if it supports that method.

Reply:
Got it; will do this tomorrow

Reply:
Upon the attempt, I cannot implement the getEntityId methods on the abstract class, because `id` is defined on the child classes. Not able to change that without either typing or naming being weird for `id` thereafter.

The other suggestion, re getEntityIdString(), is complete.